### PR TITLE
GS metadata template updates

### PR DESCRIPTION
**Pull request by tlssuse**
Hi, @chabowski, @bwgartner, @tomschr, @janajaeger:
In this PR I have updated the SUSE TRD getting started templates to include the new/updated metadata tags we have discussed.
This PR also includes updates to the GS setup script, adding many more sanity checks and clearer instructions.

The templates can be found in the `common/templates/start` directory.
The metatags are implemented in the `_docinfo-file` with variable definitions in the `_adoc-file`.

I welcome your review and feedback.

~ Terry

### CODE REVIEW

File: common/templates/start/_adoc-file [hunk 1]
```diff
@@ -2,81 +2,117 @@
...
+//   Remove or comment out unneeded variables.
+// -
+// SUSE Components
+// -
+:scomp1-provider: SUSE
+:scomp1: Primary SUSE product short name (e.g., SLES, Rancher Prime)
+:scomp1-full: Primary SUSE product full name (e.g., SUSE Linux Enterprise Server, Rancher Prime by SUSE)
+:scomp1-version: Primary SUSE product relevant version (e.g., '15', '15SP5', '2.7') or versions (e.g., '15SP4, 15SP5', '15SP3+', '2.6+', '2.x')
+:scomp1-website: Primary SUSE product website URL
+:scomp1-docs: Primary SUSE product documentation URL
```
Comment (tomschr) [inline common/templates/start/_adoc-file:65]: What's the difference between the two? Could you maybe clarify with an example? Are both needed or only one?

File: common/templates/start/_adoc-file [hunk 2]
```diff
@@ -2,81 +2,117 @@
...
+
+// -
+// Partner/Project Components
+// -
+:pcomp1-provider: Partner/Project name
+:pcomp1: Partner/Project component short name
+:pcomp1-full: Partner/Project component full name
+:pcomp1-version: Partner/Project component version or versions
+:pcomp1-page: Partner/Project component website URL
```
Comment (tomschr) [inline common/templates/start/_adoc-file:74]: Here you name it "pcomp1-page", but some lines above you have "scomp1-website". Are both semantically the same? If yes, I would suggest to use _one_ suffix to stay consistent.

File: common/templates/start/_adoc-file [hunk 3]
```diff
@@ -2,81 +2,117 @@
...
-// document
+// DOCUMENT DESCRIPTIONS
+//   Provide information to identify the guide by its purpose,
+//     use case, value.
 // -
-:title: Title
-:subtitle: Subtitle
+:usecase: (<55 characters) (e.g., Database-as-a-Service, edge analytics in healthcare, Kubernetes-native object storage)
+
+:executive-summary: Give a brief summary of what this document provides (e.g., "Streamline deployment and management of multi-tenant Kubernetes landscapes with {pcomp1-full} and {scomp1-full}.")
```
Comment (tomschr) [inline common/templates/start/_adoc-file:85]: As I found out, this goes into an `abstract/para` element. Not sure if we should add a hint about how many characters are useful. As this is not really a metadata (it's free text) there are no limits.

I just mention it here as I stumbled upon this and thought "hmn, are there no character limits?". :wink: I was first a bit confused as the other variables contains this note.

File: common/bin/gssetup.sh [hunk 4]
```diff
@@ -56,77 +45,126 @@ usecase=""
...
+  echo "  - '$PWD'"
+  echo "  -"
+  echo "  - Be sure you change to the 'kubernetes/start' or"
+  echo "  - 'linux/start' subdirectory, then execute this"
+  echo "  - script again."
+  echo "  - - - - - - - - - - - - - - - - - - - - - - - - - - - -"
+  echo
+  exit 2
+else
+  category=$(dirname $PWD)
```
Comment (tomschr) [inline common/bin/gssetup.sh:80]: There is an internal Shell abbreviation which does the same, but without executing a command:

```suggestion
  category=${PWD%/*}
```

**Review by tomschr — COMMENTED** [review]
I had a brief look and added some minor comments. Otherwise quite good! :+1:

File: common/templates/start/_adoc-file (hunk 1 above)
Comment (chabowski) [reply common/templates/start/_adoc-file:65]: Products vs Web site? suse.com/products the first, documentation.suse.com the latter? At least I assume :-)

File: common/templates/start/_adoc-file (hunk 2 above)
Comment (chabowski) [reply common/templates/start/_adoc-file:74]: does the `p` stand for partner, and the `s` for SUSE? That would make sense. Only question would be: do we need to split it down in that much detail and differ between partner and SUSE components? If this is important, fine for me - just asking :-).

File: common/templates/start/_adoc-file (hunk 1 above)
Comment (tlssuse) [reply common/templates/start/_adoc-file:65]: Good catch, @tomschr .
There is no semantic difference.
I had been going back and forth as to how to express the same thing, and I didn't synchronize before submitting.
I'll fix this in the next revision.

File: common/templates/start/_adoc-file (hunk 2 above)
Comment (tlssuse) [reply common/templates/start/_adoc-file:74]: You are correct.
There is no semantic difference.

File: common/templates/start/_adoc-file (hunk 3 above)
Comment (tlssuse) [reply common/templates/start/_adoc-file:85]: Yes, that is a bit hard to read.
I should move the instructions to a free-standing comment, perhaps.

File: common/bin/gssetup.sh (hunk 4 above)
Comment (tlssuse) [reply common/bin/gssetup.sh:80]: Thank you.
I was torn on which approach to use.
The shell shortcut is clear enough to me, but I worried about others who might need to edit the script.
I think you've convinced me to switch this in the next revision.

**Comment by chabowski** [comment]
So @tlssuse I assume I should wait with merging until the changes are done? Thank you

**Comment by tlssuse** [comment]
Yes, please hold on merging.
I will submit changes later today.

**Comment by tomschr** [comment]
> Yes, please hold on merging. I will submit changes later today.

@tlssuse As a tip: when I create a pull request and it's not ready yet, I set it to draft. It's a bit hidden, but you can find it in the right sidebar under the list of reviewers. Look for "Still in progress?" and click the "Convert to draft" link. Done. :wink:

**Comment by tlssuse** [comment]
Thank you, @tomschr ... I was wondering where to find the 'draft' option.
Done.

**Comment by tlssuse** [comment]
@chabowski: I have made some changes based on the feedback I received (thank you).

I expect to continue to evolve these files, but, for now, I think these can be merged into upstream/main.

~ Terry

**Review by chabowski — APPROVED** [review]
@tlssuse thanks for the updates! Merging now, doc should be republished today or tomorrow. :-)

**Pull request merged**